lambda/variables.tf

```diff
@@ -0,0 +1,90 @@
+variable "function_name" {
+  description = "The name of the lambda function"
+}
+
+variable "handler" {
+  description = "The lambda function handler"
+}
+
+variable "kms_key_arn" {
+  description = "The kms key to use to encrypt environment variables if necessary"
+  default     = ""
+}
+
+variable "encrypted_env_vars" {
+  default     = {}
+  type        = map(string)
+  description = "A map of environment variables in the form ENV_VAR_NAME -> plaintext value These will be encrypted"
+}
+
+variable "plaintext_env_vars" {
+  default     = {}
+  type        = map(string)
+  description = "A map of environment variables in the form ENV_VAR_NAME -> plaintext value These will not be encrypted"
+}
+
+variable "runtime" {
+  description = "The lambda runtime, for example java11"
+}
+
+variable "timeout_seconds" {
+  default = 15
+}
+
+variable "memory_size" {
+  default = 1024
+}
+
+variable "reserved_concurrency" {
+  default = 1
+}
+
+variable "tags" {}
+
+variable "efs_access_points" {
+  type = list(object({
+    access_point_arn = string,
+    mount_path       = string
+  }))
+  default     = []
+  description = "A list of access point arns and mount paths. This can be ommitted if EFS is not needed"
+}
+
+variable "vpc_config" {
+  type = list(object({
+    subnet_ids         = list(string)
+    security_group_ids = list(string)
+  }))
+}
```

Comment:
I don't think a lambda can be part of multiple VPCs at the same time, what's the use case for this being a list?

Reply:
It's there because I was using a dynamic block to set the VPC so if you didn't want a VPC, you didn't have to. Dynamic blocks need a foreach which works on lists so I had a list. The idea being that you never actually pass more than one thing in the list.
I've just found out that if you pass in empty subnets and security groups, it doesn't set the VPC so I've set them to be empty by default, changed the type to just an object and removed the dynamic VPC config which is a lot tidier.